tests/Export/ExportItemAdapterTest.php

```diff
@@ -0,0 +1,179 @@
+
+use Exception;
+use FINDOLOGIC\Export\Exceptions\EmptyValueNotAllowedException;
+use FINDOLOGIC\Export\XML\XMLItem;
+use FINDOLOGIC\FinSearch\Exceptions\Export\Product\ProductHasNoCategoriesException;
+use FINDOLOGIC\FinSearch\Export\Adapters\AdapterFactory;
+use FINDOLOGIC\FinSearch\Export\Adapters\NameAdapter;
+use FINDOLOGIC\FinSearch\Export\DynamicProductGroupService;
+use FINDOLOGIC\FinSearch\Export\ExportContext;
+use FINDOLOGIC\FinSearch\Export\ExportItemAdapter;
+use FINDOLOGIC\FinSearch\Tests\Traits\DataHelpers\ProductHelper;
+use FINDOLOGIC\FinSearch\Tests\Traits\DataHelpers\SalesChannelHelper;
+use Monolog\Logger;
+use PHPUnit\Framework\TestCase;
+use Shopware\Core\Content\Category\CategoryEntity;
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Test\TestCaseBase\IntegrationTestBehaviour;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Throwable;
+
+class ExportItemAdapterTest extends TestCase
+{
+    use IntegrationTestBehaviour;
+    use SalesChannelHelper;
+    use ProductHelper;
+
+    private $loggerMock = null;
+
+    protected function setUp(): void
+    {
+        $this->salesChannelContext = $this->buildSalesChannelContext();
+        $this->getContainer()->set('fin_search.sales_channel_context', $this->salesChannelContext);
+        $this->getContainer()->set(
+            'fin_search.export_context',
+            new ExportContext(
+                'ABCDABCDABCDABCDABCDABCDABCDABCD',
+                [],
+                $this->getCategory()
+            )
+        );
+        DynamicProductGroupService::getInstance(
+            $this->getContainer(),
+            $this->getContainer()->get('serializer.mapping.cache.symfony'),
+            Context::createDefaultContext(),
+            'ABCDABCDABCDABCDABCDABCDABCDABCD',
+            0
+        );
+
+        $this->loggerMock = $this->getMockBuilder(Logger::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        parent::setUp();
+    }
+
+    public function testProductInvalidExceptionIsLogged(): void
+    {
+        $id = Uuid::randomHex();
+
+        $productEntity = $this->createTestProduct([
+            'id' => $id,
+            'categories' => []
+        ]);
+
+        $expectedMessage = sprintf(
+            'Product "%s" with id %s was not exported because it has no categories assigned',
+            $productEntity->getTranslation('name'),
+            $productEntity->getId()
+        );
+
+        $this->expectAdapterException(
+            $productEntity,
+            new ProductHasNoCategoriesException($productEntity),
+            $expectedMessage
+        );
+    }
+
+    public function testEmptyValueIsNotAllowedExceptionIsLogged(): void
+    {
+        $id = Uuid::randomHex();
+
+        $productEntity = $this->createTestProduct([
+            'id' => $id
+        ]);
```

Comment:
The ID is never used, as you always get it with `$productEntity->getId()` anyway
```suggestion
         $productEntity = $this->createTestProduct();
```